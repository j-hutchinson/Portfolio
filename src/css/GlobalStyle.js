import { createGlobalStyle } from "styled-components"

export const GlobalStyles = createGlobalStyle`
    main {
      background: ${({ theme }) => theme.background};
      transition: all 0.5s linear;
    }

    header {
      background: ${({ theme }) => theme.primaryBackground};
      transition: all 0.5s linear;
    }

    body {
      background: ${({ theme }) => theme.background};
      color: ${({ theme }) => theme.fontColor};
      font-family: 'SF Pro Display', -apple-system, system-ui, BlinkMacSystemFont, "Segoe UI", Roboto, "Helvetica Neue", Arial, sans-serif;
      transition: all 0.5s linear;
      margin: 0;
    }

    section {
      background: ${({ theme }) => theme.sectionBackground}; 
      transition: all 0.5s linear;
    }

    a {
      color: ${({ theme }) => theme.highlight};
    }

    i {
      border: 2px solid ${({ theme }) => theme.highlight};
      transition: all 0.5s linear;
    }

    .logo {
      fill:  ${({ theme }) => theme.svgFill}; 
      transition: all 0.5s linear;
    }

    .company-link {
      color: ${({ theme }) => theme.fontColor};
    }
    
    .company-link:hover {
      color: ${({ theme }) => theme.highlight};
      transition: all 0.2s linear;
    }
`;
