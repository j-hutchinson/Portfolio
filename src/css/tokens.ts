import { css } from 'styled-components';

export const SectionStyles = css`
    border-radius: 20px;
	box-shadow: 20px 20px 60px rgb(0 0 0 / 50%), -20px -20px 60px rgb(255 255 255 / 5%);
	padding: 64px 48px;
	width: 900px;
	margin: 80px auto;

	@media screen and (max-width: 900px) {
		padding: 40px 20px;
		width: 560px;
	}

	@media screen and (max-width: 600px) {
		padding: 40px 20px;
		width: 335px;
	}
`;

export const darkTheme = {
	fontColor: 'white',
	primaryBackground: '#222',
	background: 'linear-gradient(to bottom right, #383838, #1E1E1E)',
	sectionBackground: 'radial-gradient(118.99% 670.46% at -7.06% -9.5%,#191919 0%,#0a0a0a 100%)',
	secondaryBackground: 'radial-gradient(118.99% 670.46% at -7.06% -9.5%,#191919 0%,#0a0a0a 100%)',
	border: '#EDEFE7',
	svgFill: 'white',
	highlight: '#5198db',
}

export const lightTheme = {
	fontColor: 'black',
	primaryBackground: '#eee',
	background: 'linear-gradient(to bottom right, #EDEBEF, #A6A7A4)',
	sectionBackground: 'radial-gradient(118.99% 670.46% at -7.06% -9.5%,#FFFFFF 0%,#F1F1F1 100%)',
	secondaryBackground: 'radial-gradient(118.99% 670.46% at -7.06% -9.5%,#FFFFFF 0%,#F1F1F1 100%)',
	border: '#32342B',
	svgFill: 'black',
	highlight: '#3B68A9'
}

export const colours = {
	hover: '#78A5CE',
	jobTitle: '#878787',
}