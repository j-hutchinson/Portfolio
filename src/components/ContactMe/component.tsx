import styled from 'styled-components';
import { SectionStyles } from '../../css/tokens';
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome"
import { faBeer, faChartLine, faLaptopCode } from "@fortawesome/free-solid-svg-icons"

const StyledWrapper = styled.section`
  ${SectionStyles};
`;

const StyledHeading = styled.h2`
  text-align: center;
  font-size: 32px;

  @media screen and (max-width: 600px) {
    font-size: 22px;
}
`;

const StyledText = styled.p`
  font-size: 22px;
  font-weight: 300;
  text-align: center;

  @media screen and (max-width: 600px) {
    font-size: 18px;
  }
`;

const StyledGroup = styled.div`
  display: flex;
  justify-content: center;
  margin: 0 auto;

  @media screen and (max-width: 600px) {
    margin: 24px 0;
  }
`;

const StyledGroupItem = styled.span`
  margin-right: 32px;

  :last-child {
    margin-right: 0px;
  }
`;

const ContactMe = (): JSX.Element => (
  <StyledWrapper >
    <StyledHeading>Get in touch!</StyledHeading>
    <StyledText>
      Feel free to reach out to me via any of my social medias. I want to chat to like minded people about tech, crypto or whatever you can think of.
    </StyledText>
    <StyledGroup>
      <StyledGroupItem>
        <FontAwesomeIcon icon={faLaptopCode} size="3x" />
      </StyledGroupItem>
      <StyledGroupItem>
        <FontAwesomeIcon icon={faChartLine} size="3x" />
      </StyledGroupItem>
      <StyledGroupItem>
        <FontAwesomeIcon icon={faBeer} size="3x" />
      </StyledGroupItem>
    </StyledGroup>

  </StyledWrapper >
);

export default ContactMe;
