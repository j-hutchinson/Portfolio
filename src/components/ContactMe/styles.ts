import styled from 'styled-components'
import { tokens } from '../../css/tokens';

export const StyledContainer = styled.section`
    border-radius: 20px;
    box-shadow: 20px 20px 60px rgb(0 0 0 / 20%), -20px -20px 60px rgb(255 255 255 / 5%);
    display: flex;
    justify-content: space-between;
    gap: 32px;
    padding: 64px 48px;
    width: 900px;
    margin: 80px auto;

    :first-child {
        margin-top: -60px;
    }

    @media screen and (max-width: 1100px) {
        padding: 40px 20px;
        width: 760px;
    }

    @media screen and (max-width: 900px) {
        width: 560px;
    }
   
    @media screen and (max-width: 600px) {
        width: 335px;
    }

    @media screen and (max-width: 350px) {
        :first-child {
            margin-top: 0px;
        }
    }
`;

export const StyledTitle = styled.h3`
    margin: 0px;
    font-size: ${tokens.titleFontSize};
    display: inline;

    @media screen and (max-width: 600px) {
        font-size: 18px;
    }
`;

export const StyledDescription = styled.p`
    display: block;
    font-size: 14px;
    margin: 8px 0px 20px 0px;
    line-height: 1.47059;
    font-weight: 300;
    margin-bottom: 0;
`;

export const StyledLink = styled.a`
    color: #EEE;
    background: #363636;
    border-radius: 24px;
    box-shadow: -4px -4px 10px rgb(255 255 255 / 8%), 4px 4px 10px rgb(0 0 0 / 50%);
    display: flex;
    justify-content: center;
    font-size: 12px;
    text-transform: uppercase;
    font-weight: bold;
    letter-spacing: 1.2px;
    line-height: 40px;
    margin: 24px 24px 0 0;
    padding: 4px 32px;
    text-decoration: none;
    width: 90px;

    :hover {
        text-decoration: none;
    }

    @media screen and (max-width: 900px) {
        margin: 24px 0px 0px;
    }
`;

export const StyledLeft = styled.div`
    display: flex;
    flex-direction: column;
    justify-content: space-between;
`;
