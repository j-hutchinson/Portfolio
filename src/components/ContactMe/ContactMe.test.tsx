import { shallow } from 'enzyme';
import { darkTheme } from '../../css/tokens';
import ContactMe from './component';

describe('ContactMe component', () => {
    test('component matches snapshot', () => {
        expect.assertions(1);

        const wrapper = shallow(<ContactMe theme={darkTheme} />);

        expect(wrapper).toMatchSnapshot();
    });
});