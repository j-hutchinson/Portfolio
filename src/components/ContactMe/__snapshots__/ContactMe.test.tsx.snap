// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactMe component component matches snapshot 1`] = `
<styled.section>
  <styled.h2>
    Get in touch!
  </styled.h2>
  <styled.p>
    Feel free to reach out to me via any of my social medias. I want to chat to like minded people about tech, crypto or whatever you can think of.
  </styled.p>
  <styled.div>
    <styled.span>
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              640,
              512,
              Array [],
              "f5fc",
              "M255.03 261.65c6.25 6.25 16.38 6.25 22.63 0l11.31-11.31c6.25-6.25 6.25-16.38 0-22.63L253.25 192l35.71-35.72c6.25-6.25 6.25-16.38 0-22.63l-11.31-11.31c-6.25-6.25-16.38-6.25-22.63 0l-58.34 58.34c-6.25 6.25-6.25 16.38 0 22.63l58.35 58.34zm96.01-11.3l11.31 11.31c6.25 6.25 16.38 6.25 22.63 0l58.34-58.34c6.25-6.25 6.25-16.38 0-22.63l-58.34-58.34c-6.25-6.25-16.38-6.25-22.63 0l-11.31 11.31c-6.25 6.25-6.25 16.38 0 22.63L386.75 192l-35.71 35.72c-6.25 6.25-6.25 16.38 0 22.63zM624 416H381.54c-.74 19.81-14.71 32-32.74 32H288c-18.69 0-33.02-17.47-32.77-32H16c-8.8 0-16 7.2-16 16v16c0 35.2 28.8 64 64 64h512c35.2 0 64-28.8 64-64v-16c0-8.8-7.2-16-16-16zM576 48c0-26.4-21.6-48-48-48H112C85.6 0 64 21.6 64 48v336h512V48zm-64 272H128V64h384v256z",
            ],
            "iconName": "laptop-code",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size="3x"
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
    </styled.span>
    <styled.span>
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              512,
              512,
              Array [],
              "f201",
              "M496 384H64V80c0-8.84-7.16-16-16-16H16C7.16 64 0 71.16 0 80v336c0 17.67 14.33 32 32 32h464c8.84 0 16-7.16 16-16v-32c0-8.84-7.16-16-16-16zM464 96H345.94c-21.38 0-32.09 25.85-16.97 40.97l32.4 32.4L288 242.75l-73.37-73.37c-12.5-12.5-32.76-12.5-45.25 0l-68.69 68.69c-6.25 6.25-6.25 16.38 0 22.63l22.62 22.62c6.25 6.25 16.38 6.25 22.63 0L192 237.25l73.37 73.37c12.5 12.5 32.76 12.5 45.25 0l96-96 32.4 32.4c15.12 15.12 40.97 4.41 40.97-16.97V112c.01-8.84-7.15-16-15.99-16z",
            ],
            "iconName": "chart-line",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size="3x"
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
    </styled.span>
    <styled.span>
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              448,
              512,
              Array [],
              "f0fc",
              "M368 96h-48V56c0-13.255-10.745-24-24-24H24C10.745 32 0 42.745 0 56v400c0 13.255 10.745 24 24 24h272c13.255 0 24-10.745 24-24v-42.11l80.606-35.977C429.396 365.063 448 336.388 448 304.86V176c0-44.112-35.888-80-80-80zm16 208.86a16.018 16.018 0 0 1-9.479 14.611L320 343.805V160h48c8.822 0 16 7.178 16 16v128.86zM208 384c-8.836 0-16-7.164-16-16V144c0-8.836 7.164-16 16-16s16 7.164 16 16v224c0 8.836-7.164 16-16 16zm-96 0c-8.836 0-16-7.164-16-16V144c0-8.836 7.164-16 16-16s16 7.164 16 16v224c0 8.836-7.164 16-16 16z",
            ],
            "iconName": "beer",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size="3x"
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
    </styled.span>
  </styled.div>
</styled.section>
`;
